packages/salesforcedx-vscode-core/src/commands/forceApexTestRunCodeAction.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { nls } from '../messages';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+export class ForceApexTestRunCodeActionExecutor extends SfdxCommandletExecutor<{}> {
+  private test: string;
+
+  public constructor(test: string) {
+    super();
+    this.test = test || '';
+  }
+
+  public build(data: {}): Command {
+    // inspect test value to determine description text
+    const description = this.test.includes('.')
```

Comment:
I'm a bit nervous about this one. Does this mean if you have a `a.b` you just treat it as a test method?
What happens if the class is namespaced? then you have `namespace.classname`?

Reply:
I'll generalize and remove.